policies/permissions.md

```diff
@@ -0,0 +1,100 @@
+# Permissions across the Project
+
+This document describes what access permissions are granted to individuals in
+different [roles in the project](https://www.astropy.org/team.html#roles).
+
+Occasional *temporary* write (or higher) permissions may be granted for specific
+tasks (for example, a maintainer may be given admin rights to a repo to
+configure CI for the first time or similar). Such permissions must be done
+temporarily unless prescribed otherwise by this document.
+
+For GitHub, the
+[permissions](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/repository-roles-for-an-organization)
+are enforced by adding individuals to the GitHub team
+matching their role (for example, "Astropy Core Maintainers" or "Astropy
+website maintainers" for a core maintainer or the astropy.org team, respectively),
+a duty primarily performed at the moment by the Coordination Committee.
+Temporary permissions should instead use the "collaborator" feature on Github to
+make it clear that these permissions are temporary in nature.
+
+Regardless of access level, even if it is temporary,
+[GitHub two-factor authentication](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/about-two-factor-authentication)
```

Comment:
Doesn't GitHub now enforce this anyway?

Reply:
Not sure how fast they are rolling that out. Might as well be explicit?

Reply:
@Cadair unless you are saying that I should just enforce 2FA for everyone with access to the org and be done with it?

The option says anyone still without 2FA would be kicked out but notified. Is that too controversial?

Reply:
I think that's too controversial.  A future where gh requires 2fa for every account might not be far off,  but until they do,  I think we don't have to enforce that. We can certainly add "and we strobgly recommend 2fa for all other astropy members "

Reply:
@hamogu , I think 2FA is pretty important especially for write access or above, but my opinion might not be universal. I'll think about this. If only they tell you who is actually getting kicked out before you click that button.

Reply:
CoCo discussed this on 2023-06-26 (mainly @eteq and I) and I have asked astropy-dev mailing list about this over at https://groups.google.com/g/astropy-dev/c/qKLHcJnfoMc

Reply:
https://github.blog/2023-03-09-raising-the-bar-for-software-security-github-2fa-begins-march-13/

> Last year, we [announced our commitment](https://github.blog/2022-05-04-software-security-starts-with-the-developer-securing-developer-accounts-with-2fa/) to require all developers who contribute code on GitHub.com to enable two-factor authentication (2FA) by the end of 2023.

GitHub will be requiring people with commit access to have 2fa enabled *soon*.

Reply:
So doesn't matter then if I enable it a little early? ❓

Reply:
On the one hand it will all be resolved anyway by the end of the year, on the other this makes any argument against enabling if now even weaker.
Since Astropy as an organisation is evidently not chosen among the early (or at least not the earliest) enrollment groups, we might indeed decide that we make this requirement ourselves from some point in the nearer future. This could indeed be a good argument to start the process as soon as possible, rather than having the entire rest of members run into the one-week grace period at the same time.

Reply:
Update at https://groups.google.com/g/astropy-dev/c/qKLHcJnfoMc/m/QHDWTYRfCQAJ

Maybe I can just remove this part when I can ever get back to this PR.